backend/util/util.go

```diff
@@ -161,3 +176,81 @@ func IsAuthorized(w http.ResponseWriter, r *http.Request) (bool, error) {
 	}
 	return true, nil
 }
+
+// WriteCache writes a struct to it's associated cache file for
+// a given user
+func WriteCache(username, filename string, data interface{}) {
+	// Write to JSON file
+	dirLocation := fmt.Sprintf("../data/%s", username)
+	_ = os.Mkdir(dirLocation, 0755)
+
+	fileLocation := fmt.Sprintf("../data/%s/%s.json", username, filename)
+
+	jsonData, err := json.Marshal(data)
+	if err != nil {
+		fmt.Println("ok lads")
+		log.Fatal(err)
+	}
+	_ = ioutil.WriteFile(fileLocation, jsonData, 0777)
+}
+
+// IsValidQueryType determines if an incoming query is
+// implemented by the service
+func IsValidQueryType(query string) (string, error) {
+	validTypes := []string{"accountinfo", "pullrequestcommits",
+		"pullrequests", "issuescreated",
+		"prcontributions", "repocontribs"}
+	query = strings.ToLower(query)
+
+	for _, elem := range validTypes {
+		if query == elem {
+			return fmt.Sprintf("%s.json", elem), nil
+		}
+	}
+	return "", errors.New("Invalid query type given")
```

Comment:
:bike: Maybe it'd be better to use a map instead of iterating over the slice? Take a look at [this SO answer](https://stackoverflow.com/a/15323988/6032037).